java/io/bazel/rules/closure/webfiles/server/WebfilesServer.java

```diff
@@ -143,20 +154,20 @@ public void runForever() throws IOException, InterruptedException {
       HostAndPort address = HostAndPort.fromParts(bind.getHost(), socket.getLocalPort());
       synchronized (this) {
         checkState(actualAddress == null);
-        actualAddress = address;
+        actualAddress = NetworkUtils.createUrlAddress(address);
         notify();
+        logger.info(
```

Comment:
Let's continue keeping this in the earlier place, after `synchronized(this){}`

Reply:
actualAddress is `@GuardedBy("this")`, so can only be used inside `synchronized(this){}`.

the logging uses actualAddress directly. I can either
* keep it in `synchronized(this){}` or 
* use `NetworkUtils.createUrlAddress(address)` instead.